brave_domains/BUILD.gn

```diff
@@ -6,6 +6,12 @@
 import("//build/buildflag_header.gni")
 import("buildflags.gni")
 
+if (is_official_build) {
+  # We must have a value for at least production, otherwise
+  # network requests will be sent to the subdomains as fully-qualified.
+  assert(brave_services_production_domain != "")
```

Comment:
we should also have something like 
```
#if defined(OFFICIAL_BUILD)
static_assert(!std::string(BUILDFLAG(BRAVE_SERVICES_STAGING_DOMAIN)).empty());
...
#endif
``` 
in service_domains.cc

Reply:
sorry, actually it should check that it's a valid domain in the cpp code, not check for an empty string which we have already done in gn

Reply:
not sure if there's a good way to static_assert that it's a valid domain so we might have to fall back to checking in `GetServicesDomain`